src/fosslight_source/run_scanoss.py

```diff
@@ -0,0 +1,96 @@
+# import getopt
+import logging
+# import yaml
+import json
+# import copy
+# from scancode import cli
+from datetime import datetime
+import fosslight_util.constant as constant
+from fosslight_util.set_log import init_log
+# from fosslight_util.timer_thread import TimerThread
+# from ._parsing_scancode_file_item import parsing_file_item
+# from ._parsing_scancode_file_item import get_error_from_header
+# from ._help import print_help_msg_source
+# from ._license_matched import get_license_list_to_print
+from fosslight_util.output_format import check_output_format  # , write_output_file
+from ._parsing_scanoss_file import parsing_scanResult  # scanoss
+# from ._scan_item import ScanItem
+# from .run_scancode import run_scan
+
+logger = logging.getLogger(constant.LOGGER_NAME)
+warnings.filterwarnings("ignore", category=FutureWarning)
+_PKG_NAME = "fosslight_source"
+
+
+def run_scanoss_py(path_to_scan, output_file_name="", format="", called_by_cli=False, write_json_file=False):
+    """
+    Run scanoss.py for the given path.
+
+    Args:
+        path_to_scan: path of sourcecode to scan.
+        output_file_name: file name for the output.
+
+    Returns:
+        list of ScanItem (scanned result by files) is returned.
+    """
+    if not called_by_cli:
+        global logger
+
+    _result_log = {}
+
+    scanoss_file_list = []
+    scan_command = "scanoss-py scan -o "
+
+    start_time = datetime.now().strftime('%Y%m%d_%H%M%S')
+
+    success, msg, output_path, output_file, output_extension = check_output_format(output_file_name, format)
+    if not called_by_cli:
+        logger, _result_log = init_log(os.path.join(output_path, "fosslight_src_log_"+start_time+".txt"),
+                                       True, logging.INFO, logging.DEBUG, _PKG_NAME, path_to_scan)
+
+    if output_file_name == "":
+        output_file = "SCANOSS_FOSSLight-Report_" + start_time + ".json"
+        # output_dir = os.getcwd()
+    else:
+        output_file = output_file_name + ".json"
+        # output_dir = os.path.dirname(os.path.abspath(output_file_name))
+
+    scan_command += output_file + " " + path_to_scan
+    # print(scan_command)
+    try:
+        os.system(scan_command)
+        st_json = open(output_file, "r")
+        logger.info("SCANOSS Start parsing " + path_to_scan)
+        with open(output_file, "r") as st_json:
+            st_python = json.load(st_json)
+            rc, scanoss_file_list = parsing_scanResult(st_python)
+    except Exception as error:
+        logger.warning("Parsing " + path_to_scan + ":" + str(error))
+    logger.info("|---Number of files detected with SCANOSS: " + str(len(scanoss_file_list)))
+
+    if not write_json_file:
+        try:
+            scan_command = "rm " + output_file
+            os.system(scan_command)
+            scan_command = "rm scanner_output.wfp"
+            os.system(scan_command)
+        except Exception as error:
+            logger.warning("Deleting scanoss result failed.:" + str(error))
```

Comment:
Let's change to logger.debug().
A warning is something that the user needs to check.

Reply:
done